src/eventSchemas/commands/downloadSchemaItemCode.ts

```diff
@@ -313,27 +322,53 @@ export class CodeExtractor {
         return tempFolder
     }
 
-    // Ensure that the downloaded code hierarchy has no collisions with the destination directory
-    public validateNoFileCollisions(codeZipFile: string, destinationDirectory: string): void {
+    // Check if downloaded code hierarchy has collisions with the destination directory and display them in output channel
+    public checkFileCollisions(
+        codeZipFile: string,
+        destinationDirectory: string,
+        outputChannel: vscode.OutputChannel
+    ): boolean {
         const zip = new admZip(codeZipFile)
         const zipEntries = zip.getEntries()
+        let collisionExist = false
 
         zipEntries.forEach(function(zipEntry) {
             if (zipEntry.isDirectory) {
                 // Ignore directories because those can/will merged
             } else {
                 const intendedDestinationPath = path.join(destinationDirectory, '/', zipEntry.entryName)
                 if (fs.existsSync(intendedDestinationPath)) {
-                    throw new UserNotifiedError(
-                        localize(
-                            'AWS.message.error.schemas.downloadCodeBindings.timeout',
-                            'Unable to place schema code in workspace because there is already a file {0} in the folder hierarchy',
-                            zipEntry.name
+                    if (!collisionExist) {
+                        collisionExist = true
+                        outputChannel.show(true)
+                        outputChannel.appendLine(
+                            localize(
+                                'AWS.message.info.schemas.downloadCodeBindings.colliding_files',
+                                'Following files already exist in the folder hierarchy :'
+                            )
                         )
-                    )
+                    }
+                    outputChannel.appendLine(intendedDestinationPath)
                 }
             }
         })
+
+        return collisionExist
+    }
+
+    public async overrideOnCollisionUserResponse(): Promise<boolean> {
+        const responseYes: string = localize('AWS.generic.response.yes', 'Yes')
+        const responseNo: string = localize('AWS.generic.response.no', 'No')
+
+        const userResponse = await vscode.window.showInformationMessage(
```

Comment:
These prompts are not modal, meaning the prompt can get stacked (in the bell icon in the bottom corner) if the user does not interact with it immediately. The problem is when users revisit the stack of notifications, and then respond to something like this later on. The resulting actions may be more destructive than intended (for example, if they have made changes in the destination folder.

Switching this to a modal prompt is more ideal in this situation, however I don't know if that prevents users from being able to see/scroll through the output channel (particularly if it produces a long list). Can you please investigate?

Reply:
Yes, definitely modal prompt is a better fit in this situation. It does prevent the user from scrolling through the output channel when the list is long, but that only happens when the user downloads same code bindings to the workspace for specific schemas such as AWSAPICallViaCloudTrail. Majority of schemas don't produce long list of files, so they can fit to the output channel even when downloaded second time. 

I personally would still like to go with modal prompt, I have been using the feature locally and since switching it to a modal, I can tell it is way more convenient compared to notification on the lower bottom portion. Let me know your thoughts @awschristou

Reply:
Thanks for looking into this. I think the modal makes sense given the two options. If the output channel list is large, users can cancel out, look at the list, then choose to re-select the download action.